src/test/java/io/datanerds/avropatch/PatchTest.java

```diff
@@ -4,36 +4,104 @@
 import com.google.common.collect.ImmutableList;
 import com.google.common.testing.EqualsTester;
 import io.datanerds.avropatch.operation.Operation;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.ExpectedException;
 
+import java.util.Date;
 import java.util.List;
 import java.util.Map;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
 import static io.datanerds.avropatch.operation.OperationGenerator.randomOperation;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.*;
 
 public class PatchTest {
 
-    private final Map<String, ?> headers = ImmutableMap.of("test 1", "bla", "test 2", UUID.randomUUID());
+    private final UUID uuid = UUID.randomUUID();
+    private final Map<String, ?> headers = ImmutableMap.of("test 1", "foo", "test 2", uuid);
     private final List<Operation> operations = ImmutableList.of(randomOperation(), randomOperation(), randomOperation());
+    private final UUID resource = UUID.randomUUID();
+
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
 
     @Test
     public void equality() {
         new EqualsTester()
-                .addEqualityGroup(new Patch(operations, headers), new Patch(operations, headers))
-                .addEqualityGroup(new Patch(operations))
-                .addEqualityGroup(new Patch(ImmutableList.of(randomOperation()), headers))
-                .addEqualityGroup(new Patch(ImmutableList.of(randomOperation())))
+                .addEqualityGroup(new Patch<>(resource, operations, headers), new Patch<>(resource, operations, headers))
+                .addEqualityGroup(new Patch<>(resource, operations))
+                .addEqualityGroup(new Patch<>(resource, ImmutableList.of(randomOperation()),
+                        ImmutableMap.of("test 1", "bar", "test 2", uuid)))
+                .addEqualityGroup(new Patch<>(resource, ImmutableList.of(randomOperation())))
+                .addEqualityGroup(new Patch<>(resource, ImmutableList.of(randomOperation())))
                 .testEquals();
     }
 
+    @Test
+    public void getHeader() {
+        Patch<?> patch = new Patch<>(resource, operations, headers);
+        String stringHeader = patch.getHeader("test 1");
+        assertThat(stringHeader, is(equalTo("foo")));
```

Comment:
In general there should be only one assertion per test method to avoid hiding further assertion failures after the first failing assertion which effectively may hide the real severity of the problem.

See http://blog.jayfields.com/2007/06/testing-one-assertion-per-test.html